### PR TITLE
perf: new neovim 0.6 json api

### PR DESCRIPTION
**Pull request by pedro757**
Consider this: 

1. The new json api is apparently 2.5x faster according to the [release notes of neovim 0.6](https://github.com/neovim/neovim/commit/30fed27241cc2a8f930212375f994a7fa6c99008)
2. Only for neovim 0.6

### CODE REVIEW

**Comment by pedro757** [comment]
I'm just reading the release notes :)

**Comment by lukas-reineke** [comment]
Thank you :rocket: 

But I would like to continue support for 0.5 for now.
Can you update the PR to use the new function when in 0.6, otherwise use the old one?

I am thinking it should be initialized in `new`
something like

```lua
source.new = function()
    local json_decode = vim.fn.json_decode
    if vim.fn.has "nvim-0.6" == 1 then
        json_decode = vim.json.decode
    end
    return setmetatable({
        running_job_id = 0,
        timer = vim.loop.new_timer(),
        json_decode = json_decode,
    }, { __index = source })
end
```

Then you can do `self.json_decode` inside the complete function

**Comment by pedro757** [comment]
Yeah, it's too soon to deprecate v0.5, although plugins like `nvim-lspconfig` have deprecated already any version lower than 0.6.

I've applied your changes.

File: README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ You need to have [ripgrep](https://github.com/BurntSushi/ripgrep) installed.
 
 ## Install
 
+> Requires Neovim 0.6+
```
Comment (lukas-reineke) [inline README.md:11]: remove this please

**Review by lukas-reineke — CHANGES_REQUESTED** [review]
otherwise perfect, thank you

**Comment by pedro757** [comment]
I forgot that :G

File: README.md (hunk 1 above)
Comment (pedro757) [reply README.md]: done

**Review by lukas-reineke — APPROVED** [review]
thank you

**Pull request merged**